modules/comment/logic/commentLogic.js

```diff
@@ -1,18 +1,39 @@
-const { Comment, Profile, Tip } = require('../../../models');
+const BigNumber = require('bignumber.js');
+const { Comment, Profile } = require('../../../models');
 const NotificationLogic = require('../../notification/logic/notificationLogic');
 const cache = require('../../cache/utils/cache');
+const TipLogic = require('../../tip/logic/tipLogic');
+const MdwLogic = require('../../aeternity/logic/mdwLogic');
 const { NOTIFICATION_TYPES } = require('../../notification/constants/notification');
 
 const CommentLogic = {
   async addItem(tipId, text, author, signature, challenge, parentId) {
     const parentComment = (typeof parentId !== 'undefined' && parentId !== '')
       ? await Comment.findOne({ where: { id: parentId } }) : null;
     if (parentComment === null && typeof parentId !== 'undefined' && parentId !== '') {
-      throw Error(`Could not find parent comment with id ${parentId}`);
+      throw new Error(`Could not find parent comment with id ${parentId}`);
     }
 
-    const relevantTip = await Tip.findOne({ where: { id: tipId } });
-    if (!relevantTip) throw Error(`Could not find tip with id ${tipId}`);
+    const relevantTip = await TipLogic.fetchTip(tipId);
+    if (!relevantTip) throw new Error(`Could not find tip with id ${tipId}`);
+    const parsedTip = relevantTip.toJSON();
+    // if ae --> pass
+    // check if user has balance in at least one
+    if (parsedTip.aggregation.totalAmount === '0' && parsedTip.aggregation.totalTokenAmount.length > 0) {
+      // get balances for user on all tokens
+      let foundTokenWithBalance = false;
+      // eslint-disable-next-line no-restricted-syntax
+      for (const { token } of parsedTip.aggregation.totalTokenAmount) {
```

Comment:
maybe use `asyncMap` and `.some` to match the otherwise more functional code style

Reply:
This is explicitly to avoid going through all tokens. Otherwise we could `.map` & `Promise.all`

Reply:
maybe we can use reduce then to avoid mutation and a more functional style, or just early return without the need to use `foundTokenWithBalance` at all

Reply:
mdw can also give all contract balances at once, maybe we should check this endpoint, then filter from there

https://github.com/aeternity/ae_mdw#aex-9-contract-balances-for-account

Reply:
Sounds good. Will adjust